cloudimg/CentOS-7-x86_64-Azure.ks

```diff
@@ -82,73 +94,81 @@ UPDATEDEFAULT=yes
 DEFAULTKERNEL=kernel
 EOL
 
-# make sure firstboot doesn't start
-echo "RUN_FIRSTBOOT=NO" > /etc/sysconfig/firstboot
+# Rebuild grub.cfg
+grub2-mkconfig -o /boot/grub2/grub.cfg
+
+# Ensure Hyper-V drivers are built into initramfs
+echo -e "\nadd_drivers+=\"hv_vmbus hv_netvsc hv_storvsc\"" >> /etc/dracut.conf
+kversion=$( rpm -q kernel | sed 's/kernel\-//' )
+dracut -v -f "/boot/initramfs-${kversion}.img" "$kversion"
+
+# Import CentOS public key
+rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
+
+# Enable SSH keepalive
+sed -i 's/^#\(ClientAliveInterval\).*$/\1 180/g' /etc/ssh/sshd_config
+
+# Configure network
+cat << EOF > /etc/sysconfig/network-scripts/ifcfg-eth0
+DEVICE=eth0
+ONBOOT=yes
+BOOTPROTO=dhcp
+TYPE=Ethernet
+USERCTL=no
+PEERDNS=yes
+IPV6INIT=no
+NM_CONTROLLED=no
+PERSISTENT_DHCLIENT=yes
+EOF
 
-yum clean all
+cat << EOF > /etc/sysconfig/network
+NETWORKING=yes
+NOZEROCONF=yes
+HOSTNAME=localhost.localdomain
+EOF
 
-# XXX instance type markers - MUST match CentOS Infra expectation
-echo 'azure' > /etc/yum/vars/infra
```

Comment:
we should leave this in, it allows the mirrorlist redirector to send the content download urls to the best place

Reply:
nevermind, i see it being set later.